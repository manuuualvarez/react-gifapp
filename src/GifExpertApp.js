import React, { useState } from 'react'

// Components
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>Gif expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            <ol>
                {  categories.map( (category) =>  
                    <GifGrid 
                        key= { category }
                        category= {category} 
                    />
                    )   
                }
            </ol>
        </>
    )
}
