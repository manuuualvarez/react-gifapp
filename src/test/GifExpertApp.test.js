import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'; 

// Components
import { GifExpertApp } from '../GifExpertApp';

describe('Units tests for <GifExpertApp /> (Main)', () => {
    
    test('should show GifExpertApp component', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show list of categories', () => {
        const categories = ['Miami', 'Cuba', 'Paris'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        // Show items
        expect(wrapper).toMatchSnapshot();
        // Count categories
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    

})
