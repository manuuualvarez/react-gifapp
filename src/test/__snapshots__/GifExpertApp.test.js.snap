// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Units tests for <GifExpertApp /> (Main) should show GifExpertApp component 1`] = `
<Fragment>
  <h2>
    Gif expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Units tests for <GifExpertApp /> (Main) should show list of categories 1`] = `
<Fragment>
  <h2>
    Gif expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Miami"
      key="Miami"
    />
    <GifGrid
      category="Cuba"
      key="Cuba"
    />
    <GifGrid
      category="Paris"
      key="Paris"
    />
  </ol>
</Fragment>
`;
